pkg/image/resolve_test.go

```diff
@@ -317,6 +345,76 @@ func cloneTestSystemChart(t *testing.T) string {
 	return tempDir
 }
 
+func cloneTestChart(t *testing.T) string {
+	tempDir, err := ioutil.TempDir("", "chart")
+	if err != nil {
+		t.Fatalf("failed to create temp dir: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 2*time.Minute)
+	defer cancel()
+
+	gitCloneCmd := exec.CommandContext(ctx, "git", "clone",
+		"--branch", testChartBranch,
+		"--single-branch",
+		"https://github.com/rancher/charts.git",
+		tempDir,
+	)
+	fmt.Printf("Cloning charts (branch: %s) into %s\n", testChartBranch, tempDir)
+	gitCloneCmdOutput, err := gitCloneCmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("failed to clone  chart: %s, %v", string(gitCloneCmdOutput), err)
+	}
+	fmt.Printf("Cloned  charts (branch: %s) into %s\n", testChartBranch, tempDir)
+
+	gitCheckoutCmd := exec.CommandContext(ctx, "git",
+		"-C", tempDir,
+		"checkout",
+		testChartCommit,
+	)
+	fmt.Printf("Checking out charts to %s\n", testChartCommit)
+	gitCheckoutOutput, err := gitCheckoutCmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("failed to checkout chart: %s, %v", string(gitCheckoutOutput), err)
+	}
+	fmt.Printf("Checked out charts to %s\n", testChartCommit)
+
+	prepareTestCharts(t, tempDir)
+
+	return tempDir
+}
+
+func prepareTestCharts(t *testing.T, chartDir string) {
```

Comment:
unfortunate that we can't test the extractor bash script in here somehow

Reply:
I thought the same thing, but I figured this would be good enough considering the circumstances.